server/src/main/java/org/elasticsearch/plugins/PluginsLoader.java

```diff
@@ -113,30 +113,41 @@ public static LayerAndLoader ofLoader(ClassLoader loader) {
     private final Map<String, LoadedPluginLayer> loadedPluginLayers;
 
     /**
-     * Constructs a new PluginsLoader
+     * Constructs a new PluginsLoader. The set of qualified exports/opens to Java modular modules/plugins will include all qualified
+     * exports from the server module. This is the desired behaviour in production use.
+     *
+     * @param modulesDirectory The directory modules exist in, or null if modules should not be loaded from the filesystem
+     * @param pluginsDirectory The directory plugins exist in, or null if plugins should not be loaded from the filesystem
+     */
+    public static PluginsLoader createPluginsLoaderWithServerExports(Path modulesDirectory, Path pluginsDirectory) {
+        Map<String, List<ModuleQualifiedExportsService>> qualifiedExports = new HashMap<>(ModuleQualifiedExportsService.getBootServices());
+        addServerExportsService(qualifiedExports);
+        return createPluginsLoader(modulesDirectory, pluginsDirectory, qualifiedExports);
+    }
+
+    /**
+     * Constructs a new PluginsLoader. The set of qualified exports/opens to Java modular modules/plugins will not include qualified
+     * exports from the server module. This is the desired behaviour in test use.
```

Comment:
Could this comment briefly describe what this allows tests to do? It's unusual to want different behaviour in tests versus production so it probably makes sense to explain what this allows tests to do that they could not otherwise do.